ocaml/util/xapi_version.ml

```diff
@@ -43,3 +43,27 @@ let version, xapi_version_major, xapi_version_minor, git_id =
              version
           )
     )
+
+let compare_version version_a version_b =
+  let maj_a, min_a, _ =
+    try Scanf.sscanf version_a "%d.%d.%s" (fun maj min rest -> (maj, min, rest))
+    with _ ->
+      failwith
+        (Printf.sprintf "Couldn't determine xapi version for version_a: '%s'"
+           version_a
+        )
+  in
+  let maj_b, min_b, _ =
+    try Scanf.sscanf version_b "%d.%d.%s" (fun maj min rest -> (maj, min, rest))
+    with _ ->
+      failwith
+        (Printf.sprintf "Couldn't determine xapi version for version_b: '%s'"
+           version_b
+        )
+  in
+  if maj_a > maj_b || (maj_a = maj_b && min_a > min_b) then
```

Comment:
If we don't care the exact result:
```
let is_higher a b =
  match (a.maj > b.maj, a.min > b.min) with
  | true, _
  | false, true -> true
  | _ -> false
```

Reply:
an alternative:
```ocaml
let compare a b =
  let to_maj_min v = try Scanf.sscanf v "%d.%d.%s" (fun maj min rest -> (maj, min, rest))
    with _ ->
      failwith
        (Printf.sprintf "Couldn't determine xapi version: '%s'"
           origin v
        )
  in
  let maj_a, min_a = to_maj_min a in
  let maj_b, min_b = to_maj_min b in
  let (<?>) a b = if a = 0 then b else a in
  compare maj_a maj_b <?> min_a min_b <?> 0
```
See Christian's style guide: https://github.com/lindig/ocaml-style#compare-functions